packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeProofFormatService.ts

```diff
@@ -0,0 +1,309 @@
+    return [
+      PRESENTATION_EXCHANGE_PRESENTATION_PROPOSAL,
+      PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      PRESENTATION_EXCHANGE_PRESENTATION,
+    ].includes(formatIdentifier)
+  }
+
+  public async createProposal(
+    agentContext: AgentContext,
+    { proofFormats, attachmentId }: ProofFormatCreateProposalOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const pexFormat = proofFormats.presentationExchange
+    if (!pexFormat) {
+      throw new AriesFrameworkError('Missing Presentation Exchange format in create proposal attachment format')
+    }
+
+    const { presentationDefinition } = pexFormat
+
+    ps?.validatePresentationDefinition(presentationDefinition)
+
+    const format = new ProofFormatSpec({ format: PRESENTATION_EXCHANGE_PRESENTATION_PROPOSAL, attachmentId })
+
+    const attachment = this.getFormatData(presentationDefinition, format.attachmentId)
+
+    return { format, attachment }
+  }
+
+  public async processProposal(agentContext: AgentContext, { attachment }: ProofFormatProcessOptions): Promise<void> {
+    const ps = this.presentationExchangeService(agentContext)
+    const proposal = attachment.getDataAsJson<PresentationDefinition>()
+    ps.validatePresentationDefinition(proposal)
+  }
+
+  public async acceptProposal(
+    agentContext: AgentContext,
+    { attachmentId, proposalAttachment }: ProofFormatAcceptProposalOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      attachmentId,
+    })
+
+    const presentationDefinition = proposalAttachment.getDataAsJson<PresentationDefinition>()
+
+    ps.validatePresentationDefinition(presentationDefinition)
+
+    const attachment = this.getFormatData({ presentation_definition: presentationDefinition }, format.attachmentId)
+
+    return { format, attachment }
+  }
+
+  public async createRequest(
+    agentContext: AgentContext,
+    { attachmentId, proofFormats }: FormatCreateRequestOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const presentationExchangeFormat = proofFormats.presentationExchange
+
+    if (!presentationExchangeFormat) {
+      throw Error('Missing presentation exchange format in create request attachment format')
+    }
+
+    const { presentationDefinition } = presentationExchangeFormat
+
+    ps.validatePresentationDefinition(presentationDefinition)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      attachmentId,
+    })
+
+    const options = { challenge: 'TODO' }
+
+    const attachment = this.getFormatData(
+      { options, presentation_definition: presentationDefinition },
```

Comment:
It might be nice to construct this object according to an interface, so it's harder to accidentally create an incorrect message. 

```suggestion
      { options, presentation_definition: presentationDefinition } satisfies XXX,
```